lib/src/cancelable_operation.dart

```diff
@@ -418,6 +418,33 @@ class CancelableCompleter<T> {
     });
   }
 
+  /// Makes this [CancelableCompleter.operation] complete with the same result
+  /// as [result].
+  ///
+  /// If the operation of this completer is cancelled before [result] completes,
+  /// then if [propagateCancel] is set to true, [result] is also cancelled.
```

Comment:
I use "\`true\`" for boolean literals.

The "if... then if... then" can be "if ... and ..., then"
I'd probably put the `propagateCancel` clause first:

> /// If [propagateCancel] is `true` (the default), 
> /// and the [operation] of this completer is cancelled
> /// before [result] completes,
> /// then [result] is also cancelled.

Reply:
I like that phrasing. Done.